examples/userapi/04_boundary_conditions.ipynb

```diff
@@ -0,0 +1,471 @@
+    "Additionally, we will need to specify the dimensions for our `Grid`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from devito import SpaceDimension, Constant\n",
+    "\n",
+    "x = SpaceDimension(name='x', spacing=Constant(name='h_x',\n",
+    "                                              value=extent[0]/(shape[0]-1)))\n",
+    "y = SpaceDimension(name='y', spacing=Constant(name='h_y',\n",
+    "                                              value=extent[1]/(shape[1]-1)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will need to use subdomains to accomodate the modified equations in the PML regions. As `Grid` objects are immutable, we must define our subdomains beforehand."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from devito import SubDomain\n",
+    "\n",
+    "class MainDomain(SubDomain):  # Main section with no damping\n",
+    "    name = 'main'\n",
+    "\n",
+    "    def define(self, dimensions):\n",
+    "        x, y = dimensions\n",
+    "        return {x: ('middle', PMLS, PMLS), y: y}\n",
+    "\n",
+    "\n",
+    "class Left(SubDomain):  # Left PML region\n",
+    "    name = 'left'\n",
+    "\n",
+    "    def define(self, dimensions):\n",
+    "        x, y = dimensions\n",
+    "        return {x: ('left', PMLS), y: y}\n",
+    "\n",
+    "\n",
+    "class Right(SubDomain):  # Right PML region\n",
+    "    name = 'right'\n",
+    "\n",
+    "    def define(self, dimensions):\n",
+    "        x, y = dimensions\n",
+    "        return {x: ('right', PMLS), y: y}\n",
+    "\n",
+    "\n",
+    "main_domain = MainDomain()\n",
+    "left = Left()\n",
+    "right = Right()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And create the grid, attaching our subdomains:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from devito import Grid\n",
+    "\n",
+    "grid = Grid(shape=shape, extent=extent, dimensions=(x, y),\n",
+    "            subdomains=(main_domain, left, right))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can then begin to specify our problem starting with some parameters."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "density = 1.  # 1000kg/m^3\n",
+    "velocity = 4.  # km/s"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We also need a `TimeFunction` object for each of our wavefields. Note that the pressure field is split in this case: a necessity for implementing the PML boundary conditions. As particle velocity is a vector, we will choose a `VectorTimeFunction` object to encapsulate it."
```

Comment:
Why can it not be a VectorTimeFunction ? you should still be able to access the components `ux` and `uy`

Reply:
EDIT: OK, I read the stuff below now. I'm a bit confused I think

Reply:
First order acoustic is `p` TimeFunction (pressure) and `v` VectorTimeFunction (particle velocity), these two ux, uy are a bit odd. Even if you need the two components, it should be a VectorTimeFunction I think.

Reply:
Fixed this in my most recent commit. The field splitting was a crutch in the paper to get out of Fourier space, but it never detailed how they got rid of it, so I left it in in my original version